service/history/queues/executable.go

```diff
@@ -80,14 +80,18 @@ var (
 	schedulerRetryPolicy = common.CreateTaskProcessingRetryPolicy()
 	// reschedulePolicy is the policy for determine reschedule backoff duration
 	// across multiple submissions to scheduler
-	reschedulePolicy             = common.CreateTaskReschedulePolicy()
-	taskNotReadyReschedulePolicy = common.CreateTaskNotReadyReschedulePolicy()
+	reschedulePolicy                      = common.CreateTaskReschedulePolicy()
+	taskNotReadyReschedulePolicy          = common.CreateTaskNotReadyReschedulePolicy()
+	taskResourceExhuastedReschedulePolicy = common.CreateTaskNotReadyReschedulePolicy()
 )
 
 const (
 	// resubmitMaxAttempts is the max number of attempts we may skip rescheduler when a task is Nacked.
 	// check the comment in shouldResubmitOnNack() for more details
 	resubmitMaxAttempts = 10
```

Comment:
is it a bad idea to drop this down to much smaller like 3?

Reply:
Sounds reasonable to me, but I am a little bit worried about if that will increase the latency for the "one workflow, large activity/childworkflow batch" case (tasks failing with workflow busy error) too much. 
Let me add a TODO here and do some testings before reducing it.